### PR TITLE
Don't accept posts from unknown owners

### PR DESCRIPTION
**Pull request by annando**
This fixes the issue that had been described here: https://forum.friendi.ca/display/2196d4ee145b4b6c027d6da327610548

### CODE REVIEW

File: src/Protocol/DFRN.php [hunk 1]
```diff
@@ -2431,6 +2432,8 @@ private static function processEntry($header, $xpath, $entry, $importer, $xml)
 		// Fetch the owner
 		$owner = self::fetchauthor($xpath, $entry, $importer, "dfrn:owner", true);
 
+		$owner_unknown = (isset($owner["contact-unknown"]) && $owner["contact-unknown"]);
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php]: This line can be replaced by the shorter `$owner_unknown = empty($owner["contact-unknown"]);`.

File: src/Protocol/DFRN.php [hunk 2]
```diff
@@ -2659,6 +2662,13 @@ private static function processEntry($header, $xpath, $entry, $importer, $xml)
 			return;
 		}
 
+		// This check is done here to be able to receive connection requests in "processVerbs"
+		if (($entrytype == DFRN::TOP_LEVEL) && $owner_unknown) {
+			logger("Item won't be stored because user " . $importer["importer_uid"] . " doesn't follow " . $item["owner-link"] . ".", LOGGER_DEBUG);
+			return;
+		}
+
+
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php]: Code standards: Please remove an extraneous line feed.

**Comment by MrPetovan** [comment]
Ha, someone was quicker than me.

File: src/Protocol/DFRN.php (hunk 1 above)
Comment (annando) [reply src/Protocol/DFRN.php]: Wouldn't this invert the boolean value?

Comment (MrPetovan) [reply src/Protocol/DFRN.php]: You're right, it's 3:20 AM here and apparently I can't think straight. I updated my comment accordingly.

**Comment by tobiasd** [comment]
Yes, sorry. But I hope that my stream will get usable with this again.

**Comment by annando** [comment]
It should. Have a look at the log file to see if its working.

**Comment by tobiasd** [comment]
I don't need the logs for this. It already looks much better then before. I'll observe if I see any unwanted postings again, but so far, so good. I had plenty of them.

**Comment by annando** [comment]
Please have a look if you still receive comments on your own posts from persons you don't follow (you should get these) and if you see forum posts from people you don't follow. This *should* work, but I hadn't been able to test this.

**Comment by tobiasd** [comment]
As I post most postings into private groups I really hope there will be no comments by non followers ;-) Forums were rather quiet yesterday, but I keep an open eye on that.

**Pull request merged**